js/router.js

```diff
@@ -77,19 +76,65 @@ export default class ObRouter extends Router {
     });
   }
 
-  user(guid, tab, ...args) {
-    tab = tab || 'store'; // eslint-disable-line no-param-reassign
-    const pageOpts = { tab };
+  get userStates() {
+    return [
+      'home',
+      'store',
+      'following',
+      'followers',
+    ];
+  }
+
+
+  /**
+   * Based on the route arguments, determine whether we
+   * have a valid user route.
+   */
+  isValidUserRoute(guid, state, ...deepRouteParts) {
```

Comment:
I made some changes in the user page route so that it's more explicit on what are valid urls. This is something a while back I talked about with @morebrownies and he seemed to agree with me that, in general, if a user enters an unrecognized url, we take them to a Page Not Found page, instead of trying to guess where they want to go. I think it's much cleaner in that it takes out ambiguity of what a route represents.

Anyhow, before it was like this:

\<guid|handle\>/crap
\<guid|handle\>/crap/hippo

would all redirect to `<guid|handle>/store`.

Similarly:

\<guid|handle\>/home/poo
\<guid|handle\>/following/shoo

would redirect to `<guid|handle>/home` & `<guid|handle>/following` respectively.

Now, any unrecognized urls will redirect to a Page Not Found page (which obviously can use a better design):

![image](https://cloud.githubusercontent.com/assets/794517/19742682/0bf40a06-9b7b-11e6-81a9-a5c2995657b8.png)

The one exception to this is entering just the \<guid|handle\> in the address bar, which will redirect you to `
<guid|handle>/store`.